src/Microsoft.TestPlatform.Extensions.TrxLogger/XML/XmlPersistence.cs

```diff
@@ -700,13 +700,23 @@ private static string RemoveInvalidXmlChar(string str)
                 // From xml spec (http://www.w3.org/TR/xml/#charsets) valid chars: 
                 // #x9 | #xA | #xD | [#x20-#xD7FF] | [#xE000-#xFFFD] | [#x10000-#x10FFFF]  
 
-                string invalidChar = @"[^\x09\x0A\x0D\x20-\uD7FF\uE000-\uFFFD\u10000-\u10FFFF]";
-                return Regex.Replace(str, invalidChar, TrxLoggerResources.PlaceHolderForInvalidChar);
+                // we are handling only #x9 | #xA | #xD | [#x20-#xD7FF] | [#xE000-#xFFFD]
```

Comment:
add unit test string to include Unicode character valid & invalid in all of these ranges